azure/virtual_machine/variables.tf

```diff
@@ -0,0 +1,169 @@
+variable "resource_group_name" {
+  type        = string
+  description = "The name of an existing Resource Group."
+}
+
+variable "vm_name" {
+  description = "The name of the virtual machine."
+  type        = string
+}
+
+variable "environment" {
+  description = "The environment in which the resource should be provisioned."
+  type        = string
+}
+
+variable "vm_size" {
+  description = "The SKU which should be used for this Virtual Machine. For an exaustive list of virtual, please use the command 'az vm list-sizes --location 'your-location''."
+  type        = string
+  default     = "Standard_D2s_v3"
+
+  validation {
```

Comment:
In this validation you're only allowing Standard_D2s_V3, or am I seeing it wrong, shouldn't it be a predefined list?

Reply:
Yes, it should be. I'll talk to you, so we can review the list of vms allowed. Thanks for pointing out